src/rpc/mempool.cpp

```diff
@@ -888,34 +890,57 @@ static RPCHelpMan submitpackage()
             Chainstate& chainstate = EnsureChainman(node).ActiveChainstate();
             const auto package_result = WITH_LOCK(::cs_main, return ProcessNewPackage(chainstate, mempool, txns, /*test_accept=*/ false));
 
-            // First catch any errors.
+            std::map<Wtxid, std::optional<std::string>> wtixd_to_error_strings;
+            std::string package_msg = "success";
+
+            // First catch any errors and capture useful error strings.
             switch(package_result.m_state.GetResult()) {
-                case PackageValidationResult::PCKG_RESULT_UNSET: break;
-                case PackageValidationResult::PCKG_POLICY:
+                case PackageValidationResult::PCKG_RESULT_UNSET:
                 {
-                    throw JSONRPCTransactionError(TransactionError::INVALID_PACKAGE,
-                        package_result.m_state.GetRejectReason());
+                    // Everything in the package should be submitted
```

Comment:
Comment says "Everything in the package should be submitted" but you're only checking that the results exist in `m_tx_results`. Do you want to check that `mempool.exists(wtxid)` as well?

Reply:
I think the check would be for `Txid` since different witness isn't considered a failure? added

Reply:
yeah makes sense